nsqd/options.go

```diff
@@ -149,3 +156,27 @@ func NewOptions() *Options {
 		TLSMinVersion: tls.VersionTLS10,
 	}
 }
+
+func applyDefaultOptions(opts *Options) error {
```

Comment:
We don't really need a new function for this, we already have a pattern for option defaults that depend on other options:

```go
func New(opts *Options) (*NSQD, error) {
	...
	if opts.Logger == nil {
		opts.Logger = log.New(os.Stderr, opts.LogPrefix, log.Ldate|log.Ltime|log.Lmicroseconds)
	}
	...
	if opts.StatsdPrefix != "" {
		var port string
		_, port, err = net.SplitHostPort(opts.HTTPAddress)
		if err != nil {
			return nil, fmt.Errorf("failed to parse HTTP address (%s) - %s", opts.HTTPAddress, err)
		}
		statsdHostKey := statsd.HostKey(net.JoinHostPort(opts.BroadcastAddress, port))
		prefixWithHost := strings.Replace(opts.StatsdPrefix, "%s", statsdHostKey, -1)
		if prefixWithHost[len(prefixWithHost)-1] != '.' {
			prefixWithHost += "."
		}
		opts.StatsdPrefix = prefixWithHost
	}

	if opts.TLSClientAuthPolicy != "" && opts.TLSRequired == TLSNotRequired {
		opts.TLSRequired = TLSRequired
	}
	...
```

I defer to @mreiferson about whether we want to move all of that kind of logic out of `nsqd.New()` over to here, but I lean towards "nah".

Reply:
I moved it to `New()`. 

I guess shifting to new a function would be more appropriate in a different PR.

Reply:
Maybe we should also update the logic quoted above, which uses the http port and broadcast address to form a StatsdPrefix, to use the broadcast http port?